### DISCUSSION TITLE
pywraplp solution callback feature

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#3612] [CP-SAT questions] [answered]
Hi Experts,

Could I know if we have plan to support solution callback in pywraplp interface like pywrapsat.SolutionCallback? Support CP-SAT as solver would be enough, not necessary for MIP solver.

John

**Comment by lperron** [comment, marked as answer]
Why don't you use CP-SAT native api. It supports callbacks  in all languages. 

For pywraplp, it will  not happen. It will be implemented in math_opt. But python is not yet implemented in or-tools.

**Comment by C8504** [comment]
Thanks lperron.
The reason why I need this is that I am using pywraplp to solve instances of an integer programming model. 
Some instances by CP-SAT solver are faster, while some instances are faster by mip solver. So used pywarplp as modeling wrap, and switch different solvers, pywraplp is wonderful to switch solvers. Hopefully you could invest more on this interface.

Consequently， I prefer the same feature in pywraplp like pywrapsat. Thanks again.

John

**Comment by lperron** [comment]
Yes. As I said, it will not be implemented in pywraplp.

**Comment by C8504** [comment]
Thanks Perron.
